aws-lc-rs/src/rsa/key.rs

```diff
@@ -499,3 +582,102 @@ unsafe fn serialize_RSA_pubkey(pubkey: &ConstPointer<RSA>) -> Result<Box<[u8]>,
     let pubkey_vec = Vec::from(pubkey_slice);
     Ok(pubkey_vec.into_boxed_slice())
 }
+
+pub(super) fn generate_rsa_key(
+    size: std::os::raw::c_int,
+    fips: bool,
+) -> Result<LcPtr<EVP_PKEY>, Unspecified> {
+    // We explicitly don't use `EVP_PKEY_keygen`, as it will force usage of either the FIPS or non-FIPS
+    // keygen function based on the whether the build of AWS-LC had FIPS enbaled. Rather we delegate to the desired
+    // generation function.
+
+    let rsa = DetachableLcPtr::new(unsafe { RSA_new() })?;
+
+    if 1 != if fips {
+        indicator_check!(unsafe { RSA_generate_key_fips(*rsa, size, null_mut()) })
+    } else {
+        // Safety: RSA_F4 == 65537, RSA_F4 an i32 is safe to cast to u64
+        debug_assert_eq!(RSA_F4 as u64, 65537u64);
```

Comment:
Can we define RSA_F4 as a `const u64` locally to avoid the need for casting here?